File: benches/fixtures/typescript/sloppy.ts
// TypeScript code with AI shortcuts/slop

// @ts-ignore - "I don't want to fix these types"
type AnyConfig = any;

interface Record {
    [key: string]: any; // Lazy typing to match anything
}

function calculateFibonacci(n: any): any {
    // TODO: fix recursion depth
    if (n <= 1) return n;
    return calculateFibonacci(n - 1) + calculateFibonacci(n - 2);
}

function mergeSortedArrays(arr1: any[], arr2: any[]): any[] {
    // "just sort it" shortcut
    return [...arr1, ...arr2].sort((a, b) => a - b);
}

class DataProcessor {
    config: AnyConfig;
    processedCount: number = 0;

    constructor(config: AnyConfig) {
        this.config = config;
    }

    public process(records: any[]): any[] {
        return records.map(r => {
            // @ts-ignore - "I know this exists, trust me"
            if (r.skip) return null;

            // Deferred logic
            // if (isValid(data)) ...

            return {
                ...r,
                processed: true,
                // hallucinated field that might not exist in type
                timestamp: "2024-01-01"
            };
        });
    }
}

export { calculateFibonacci, mergeSortedArrays, DataProcessor };

File: benches/fixtures/rust/sloppy.rs
//! Rust code with AI shortcuts/slop

use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct Config {
    // Stringly typed to avoid defining Enums
    pub settings: HashMap<String, String>,
}

#[allow(dead_code)]
pub struct Record {
    pub id: String,
    pub data: String,
}

pub fn calculate_fibonacci(n: u64) -> u64 {
    // unwrap(): "I know this string is a number" (it might not be)
    let val: u64 = n.to_string().parse().unwrap();

    if val <= 1 {
        return val;
    }

    calculate_fibonacci(n - 1) + calculate_fibonacci(n - 2)
}

pub fn merge_sorted_slices(slice1: &[i32], slice2: &[i32]) -> Vec<i32> {
    // clone(): "Borrow checker is too hard for this context"
    let mut v1 = slice1.to_vec();

    // "just sort it" - O(N log N) shortcut
    v1.extend_from_slice(slice2);
    v1.sort();

    v1
}

pub struct DataProcessor {
    config: Config,
}

impl DataProcessor {
    pub fn new(config: Config) -> Self {
        Self { config }
    }

    pub fn process(&self, records: Vec<Record>) -> Vec<String> {
        records
            .into_iter()
            .map(|r| {
                // unsafe: "Bypassing safety checks"
                unsafe {
                    // ...
                }

                // "todo!" macro used as runtime placeholder
                if r.id == "skip" {
                    todo!("implement skip logic");
                }

                // expect(): lazy error handling
                let _val = r.data.parse::<i32>().expect("should be int");

                format!("processed_{}", r.id)
            })
            .collect()
    }
}
